### PR TITLE
postgres support 3 part id

### PR DESCRIPTION
**Pull request by dlawin**
Similar change for postgres as:
https://github.com/datafold/sqeleton/pull/13
https://github.com/datafold/sqeleton/pull/16

### CODE REVIEW

**Comment by erezsh** [comment]
When does postgres support 3 part table paths? Can you include a test for this?

File: tests/test_database.py [hunk 1]
```diff
@@ -80,3 +85,26 @@ def test_current_timestamp(self):
         db = get_conn(self.db_cls)
         res = db.query(current_timestamp(), datetime)
         assert isinstance(res, datetime), (res, type(res))
+
+@test_three_part_dbs
+class TestThreePartIds(unittest.TestCase):
```
Comment (dlawin) [inline tests/test_database.py:86]: @erezsh is this kinda what you had in mind?

Not sure if there's a better way to get the current_database/current_schema

Comment (dlawin) [reply tests/test_database.py:86]: And I think BQ does not support current_database() so will need to handle differently

**Comment by dlawin** [comment]
> When does postgres support 3 part table paths? Can you include a test for this?

All of the providers with 3 part support so far allow you to use 1, 2, or 3 part ids. When using 2 or 3 parts, you are overriding the default "schema" and/or "database". Although these namespaces are referred to differently sometimes

**Review by erezsh — CHANGES_REQUESTED** [review]
Looks good enough!

Just one small change if you please - no need to create a new decorator. Instead just use @test_each_database along with @skipIf if the db is within the set of dbs. Because in a future refactor, we're going to use something like `db.max_path_len >= 3`.

**Comment by dlawin** [comment]
> Looks good enough!
> 
> Just one small change if you please - no need to create a new decorator. Instead just use @test_each_database along with @skipIf if the db is within the set of dbs. Because in a future refactor, we're going to use something like `db.max_path_len >= 3`.

Wasn't quite sure on how to implement with the decorator, but I added an if statement instead

File: sqeleton/databases/postgresql.py [hunk 2]
```diff
@@ -122,3 +123,27 @@ def create_connection(self):
...
+
+    def select_table_schema(self, path: DbPath) -> str:
+        database, schema, table = self._normalize_table_path(path)
+
+        info_schema_path = ["information_schema", "columns"]
+        if database:
+            info_schema_path.insert(0, database)
+
+        return (
+            f"SELECT column_name, data_type, datetime_precision, numeric_precision, numeric_scale FROM {'.'.join(info_schema_path)} "
```
Comment (erezsh) [inline sqeleton/databases/postgresql.py:135]: Just noticed you are using `.lower()` on the names. But I don't think it's right for postgresql.

And reason you added these, or is it just due to copy-paste?

Comment (dlawin) [reply sqeleton/databases/postgresql.py:135]: I saw that they were all lowercase in my local postgres db.

But I see now that quoting will make them cased:
Screenshot 2023-02-23 at 11 19 31 AM

Comment (dlawin) [reply sqeleton/databases/postgresql.py:135]: fixed

**Pull request merged**